kong/llm/drivers/openai.lua

```diff
@@ -155,13 +172,13 @@ function _M.subrequest(body, conf, http_opts, return_res_table)
     headers[conf.auth.header_name] = conf.auth.header_value
   end
 
-  local res, err = ai_shared.http_request(url, body_string, method, headers, http_opts)
+  local res, err, httpc = ai_shared.http_request(url, body_string, method, headers, http_opts, return_res_table)
```

Comment:
I think the naming of this parameter is sort of a stumbling block. Elsewhere you are using `_string` and `_table` suffixes that denote the var type, but here `return_res_table` looks like it's actually a boolean. At the function definition the parameter name is `buffered` which seems more clearly a boolean (but also slightly unclear in its purpose).

BTW if you want a better experience not having to remember types passed around between functions, I would suggest annotating your code with type definitions:
https://luals.github.io/wiki/annotations/

Reply:
I'll fix the naming. This will be gone in a couple of minor versions anyway once SSE is fully supported, so it's not... massive problem.